### PR TITLE
WIP: Adds AZ Entra support to CS db

### PR DESCRIPTION
**Pull request by markturansky**
This PR adds AZ Entra authentication to CS database.

DO NOT MERGE while we work out the kinks :)

### CODE REVIEW

File: cluster-service/Makefile [hunk 1]
```diff
@@ -2,23 +2,29 @@ SHELL = /bin/bash
...
 	sed -e "s/DOMAIN/$${DOMAIN}/g" -e "s/REGION/${REGION}/g" deploy/mvp-provisioning-shards.yml > deploy/tmp-provisioning-shard.yml
 	oc process --local -f deploy/openshift-templates/arohcp-namespace-template.yml \
 	  -p ISTIO_VERSION=asm-1-20 | oc apply -f -
-	oc process --local -f deploy/openshift-templates/arohcp-db-template.yml | oc apply -f -
+
 	oc process --local -f deploy/openshift-templates/arohcp-secrets-template.yml \
-	  -p PROVISION_SHARDS_CONFIG="$$( base64 -i deploy/tmp-provisioning-shard.yml)" | oc apply -f -
+		-p DB_HOST=$(shell az postgres flexible-server list --resource-group ${RESOURCEGROUP} --query "[?starts_with(name, 'cs-pg-')].fullyQualifiedDomainName" -o tsv) \
+		-p DB_NAME=cluster-service \
+		-p DB_USER=cs-pg-aaaaaaaaaaaaa-db-admin-msi \
```
Comment (markturansky) [inline cluster-service/Makefile:16]: this is the name I see in Az portal under Entra admins.  This db username yields the error: 

` Can't connect to the database: pq: Service Principals cannot generate AAD_AUTH_TOKENTYPE_APP_USER tokens for role`

any other name (such as `clusters-service-eastus` or `cs-pg-aaaaaaaaaaaaa` (db name)) will timeout.

When I added my own user (my email at redhat) as Entra admin to this db instance, I can successfully login.

File: cluster-service/Makefile [hunk 2]
```diff
@@ -2,23 +2,29 @@ SHELL = /bin/bash
...
 	AZURE_CS_MI_CLIENT_ID=$(shell az identity show \
 			-g ${RESOURCEGROUP} \
 			-n clusters-service-${REGION} \
 			--query clientId) && \
 	oc process --local -f deploy/openshift-templates/arohcp-service-template.yml \
 	  -p AZURE_CS_MI_CLIENT_ID=$${AZURE_CS_MI_CLIENT_ID} \
 	  -p IMAGE_REGISTRY=devarohcp.azurecr.io \
-	  -p IMAGE_REPOSITORY=app-sre/uhc-clusters-service \
-	  -p IMAGE_TAG=fc17932 | oc apply -f -
+	  -p IMAGE_REPOSITORY=cluster-service/uhc-clusters-service \
```
Comment (markturansky) [inline cluster-service/Makefile:27]: this will change back to app-sre, but this is where I can push dev images.

**Comment by markturansky** [comment]
@microsoft-github-policy-service agree [company="Red Hat"]

**Comment by markturansky** [comment]
@microsoft-github-policy-service agree company="Red Hat"

File: cluster-service/Makefile (hunk 1 above)
Comment (geoberle) [reply cluster-service/Makefile:16]: Thanks for figuring this out. I guess the MI backed account lacks some permissions. I will have a look

**Comment by markturansky** [comment]
@geoberle some fixes here.  @bennerv and I figured out some db grants that need to be assigned to the role.

**Comment by markturansky** [comment]
closed in favor of https://github.com/Azure/ARO-HCP/pull/473

**Pull request closed without merging**